### PR TITLE
[nnx] improve mnist tutorial

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

Uses `.eval()` on inference in the MNIST Tutorial

### CODE REVIEW

File: uv.lock
Comment (superbobry) [inline uv.lock:1]: This seems unrelated?

File: docs_nnx/mnist_tutorial.md [hunk 1]
```diff
@@ -246,6 +246,8 @@ plt.show()
 Create a `jit`-compiled model inference function (with `nnx.jit`) - `pred_step` - to generate predictions on the test set using the learned model parameters. This will enable you to visualize test images alongside their predicted labels for a qualitative assessment of model performance.
 
 ```{code-cell} ipython3
+model.eval() # Switch to evaluation mode.
```
Comment (superbobry) [inline docs_nnx/mnist_tutorial.md:249]: Consider explaining a bit more why this is necessary.

**Review by superbobry — APPROVED** [review]

File: docs_nnx/mnist_tutorial.md (hunk 1 above)
Comment (cgarciae) [reply docs_nnx/mnist_tutorial.md:249]: Added an explanation.
